integration-tests/src/tests/runtime/sanity_checks.rs

```diff
@@ -106,5 +106,22 @@ fn test_cost_sanity() {
             "receipts_gas_profile"
         },
         receipts_gas_profile
+            .iter()
+            .map(|gas_profile| {
+                gas_profile
+                    .iter()
+                    .cloned()
+                    .map(|cost| {
+                        if cost.cost == "CONTRACT_LOADING_BYTES" {
+                            // Ignore `gas_used` of `CONTRACT_LOADING_BYTES` since contract size is
+                            // non-deterministic. Size may depend on random environmental factors.
+                            CostGasUsed { gas_used: 0, ..cost }
+                        } else {
+                            cost
+                        }
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Vec<_>>()
```

Comment:
Could you do the filtering when `receipts_gas_profile` is created? With it being here the whole expression is in the snap file which IMO is rather ugly and prone to spurious test failures when this code gets refactored in any way.